config/fields/mixins/pagepicker.php

```diff
@@ -3,23 +3,35 @@
 return [
     'methods' => [
         'pagepicker' => function (array $params = []) {
-            $query = $params['query'] ?? null;
-            $model = $this->model();
-            $site  = $this->kirby()->site();
+            $query    = $params['query'] ?? null;
+            $subpages = $params['subpages'] ?? true;
+            $model    = $this->model();
+            $site     = $this->kirby()->site();
 
-            if ($query) {
-                $pages = $model->query($query, 'Kirby\Cms\Pages');
-                $self  = null;
+            if (empty($query) === false) {
+                if ($subpages === true) {
+                    if ($parent = $site->find($params['parent'] ?? null)) {
+                        $pages = $parent->children();
+                    } else {
+                        $pages  = $model->query($query, 'Kirby\Cms\Pages');
+                        $parent = $pages->parent();
+                    }
+                } else {
+                    $pages = $model->query($query, 'Kirby\Cms\Pages');
+                }
             } else {
                 if (!$parent = $site->find($params['parent'] ?? null)) {
                     $parent = $site;
                 }
-
+                
                 $pages = $parent->children();
+            }
+            
+            if ($subpages === true) {
                 $self  = [
-                    'id'     => $parent->id() == '' ? null : $parent->id(),
-                    'title'  => $parent->title()->value(),
+                    'id'     => (empty($parent->id()) === true || empty($params['parent']) === true) ? null : $parent->id(),
```

Comment:
Why the check for the `$params['parent']` value?

Reply:
![FireShot Capture 006 - Sandbox - Mægazine - localhost](https://user-images.githubusercontent.com/3393422/66274075-5d964600-e883-11e9-8136-7f6410792414.png)

This is for hide back button for root.